dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,209 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+                boolean siteSearchSecondLanguage, boolean createContentInDefaultLanguage,
+                boolean createContentInSecondLanguage, boolean createPageInDefaultLanguage,
+                boolean createPageInSecondLanguage,
+                ExpectedResults expectedResultsWhenSearchingContentInDefaultLanguage,
+                ExpectedResults expectedResultsWhenSearchingContentInSecondLanguage) {
+            this.defaultPageToDefaultLanguage = defaultPageToDefaultLanguage;
+            this.defaultContentToDefaultLanguage = defaultContentToDefaultLanguage;
+            this.siteSearchDefaultLanguage = siteSearchDefaultLanguage;
+            this.siteSearchSecondLanguage = siteSearchSecondLanguage;
+            this.createContentInDefaultLanguage = createContentInDefaultLanguage;
+            this.createContentInSecondLanguage = createContentInSecondLanguage;
+            this.createPageInDefaultLanguage = createPageInDefaultLanguage;
+            this.createPageInSecondLanguage = createPageInSecondLanguage;
+            this.expectedResultsWhenSearchingContentInDefaultLanguage = expectedResultsWhenSearchingContentInDefaultLanguage;
+            this.expectedResultsWhenSearchingContentInSecondLanguage = expectedResultsWhenSearchingContentInSecondLanguage;
+        }
+    }
+
+    static class ExpectedResults {
+        boolean pageInDefaultLang;
+        boolean pageInSecondLang;
+
+
+        public ExpectedResults(boolean pageInDefaultLang, boolean pageInSecondLang) {
+            this.pageInDefaultLang = pageInDefaultLang;
+            this.pageInSecondLang = pageInSecondLang;
+        }
+    }
 
+    @DataProvider
+    public static Object[] siteSearchTestCases() {
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in default language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case1 = new TestCaseSiteSearchBuilder()
+                .siteSearchDefaultLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only.
+         * DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true
+         * DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in second language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case2 = new TestCaseSiteSearchBuilder()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'case2' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)